### ISSUE TITLE
Issues with List typescript definitions

### ISSUE THREAD

**Issue by BenLorantfy** [facebook/immutable-js#1514]
### What happened
Typescript errors when trying to use `.map`. 
```
    import { List } from 'immutable'; 

    const a: List<number> = List([1,2,3]);
    const b: List<number> = a.map(n => n*2);
```
- `b` gives this error: `Type 'Iterable<number, number>' is not assignable to type 'List<number>'.
  Property 'set' is missing in type 'Iterable<number, number>'.`
- `n` gives this error: `Object is possibly 'undefined'.`

### How to reproduce
Try to compile the code above.

**Comment by BenLorantfy** [comment]
nvm, I see this has been fixed in the 4.0 releases candidate.

**Comment by japser36** [comment]
A note for DraftJS users looking for this. 

Try 
```ts
const myList = List([1,2,3,4]).map(/* ... */).toList()
```

toList converts the Iterable returned from .map() back into a List :)

Since DraftJS currently still requires version immutable@~3.7.4, I ran into this issue trying to use immutable's List. It was resolving to Draft's version even if I had immutable 4.0.0 installed directly as a dependency, since DraftJS's @types/draft-js/node_modules/immutable was being resolved for the types instead of immutable 4 from my direct dependency. I tried adding typeRoots and all kinds of stuff, but it just wouldn't cooperate in the end... it would be great if Draft updated their Immutable dependency soon!
